src/solvers/mp/MasterProblem.cpp

```diff
@@ -0,0 +1,1278 @@
+	rep << "\n";
+
+	cout << rep.str();
+
+	return rep.str();
+}
+
+
+string MasterProblem::allocationToString(bool printInteger){
+	std::stringstream rep;
+
+	int nbNurses = pScenario_->nbNurses_;
+	int nbShifts = pScenario_->nbShifts_;
+	int firstDay = pDemand_->firstDay_, nbDays = pDemand_->nbDays_;
+
+	rep << std::endl;
+	rep << "Allocations of the (potentially fractional) current solution:" << std::endl;
+	rep << "\t\t  ";
+	for (int day = firstDay; day < firstDay+nbDays; day++) {
+		rep << "| " << Tools::intToDay(day).at(0) << " ";
+	}
+	rep << "|" << std::endl;
+	rep << "-------------------------------------"<< std::endl;
+
+  vector3D<double> fractionalRoster = getFractionalRoster();
+	for (int n = 0; n < nbNurses; n ++) {
+		LiveNurse* pNurse = theLiveNurses_[n];
+		const vector2D<double>& fnurseFractionalRoster = fractionalRoster[n];
+		rep << pNurse->name_ << "\t";
+		for(int s=1; s<nbShifts; ++s){
+			if (s>1) rep << "\t";
+			rep << pScenario_->intToShift_[s] << "\t";
+			for (int day = firstDay; day < firstDay+nbDays; day++){
+				double shiftValue = fnurseFractionalRoster[day][s];
+				if(shiftValue > 1-EPSILON){
+					rep << "|  1 ";
+				} else if(shiftValue > EPSILON){
+					char buffer[100];
+					sprintf(buffer, "|%1.2f", shiftValue);
+					rep << buffer;
+				} else {
+					rep << "| -- ";
+				}
+				if(Tools::isSunday(day)) rep << "| ";
+			}
+			rep << "|" << std::endl;
+		}
+		rep << std::endl;
+	}
+	rep << std::endl;
+
+	cout << rep.str();
+
+	return rep.str();
+}
+
+string MasterProblem::coverageToString(bool printInteger){
+	std::stringstream rep;
+
+	int nbShifts = pScenario_->nbShifts_;
+	int nbSkills = pScenario_->nbSkills_;
+	int firstDay = pDemand_->firstDay_, nbDays = pDemand_->nbDays_;
+
+	rep << std::endl;
+	rep << "Coverage of the (potentially fractional) current solution:" << std::endl;
+	rep << "\t\t   ";
+	for (int day = firstDay; day < firstDay+nbDays; day++) {
+		rep << "| " << Tools::intToDay(day).at(0) << " ";
+	}
+	rep << "|" << std::endl;
+	rep << "-------------------------------------"<< std::endl;
+
+	string tab = "\t";
+
+	for(int s=1; s<nbShifts; ++s){
+		rep << pScenario_->intToShift_[s] << "\t";
+		for(int sk=0; sk<nbSkills; sk++){
+			if(sk!=0) rep << "\t";
+
+			char buffer0[20];
+			string skill = pScenario_->intToSkill_[sk];
+			sprintf(buffer0, "%-12s", skill.c_str());
+			rep << buffer0;
+
+			for (int day = firstDay; day < firstDay+nbDays; day++){
+				double shiftValue = pModel_->getVarValue(skillsAllocVars_[day][s-1][sk]);
+				char buffer[100];
+				if(abs(shiftValue - round(shiftValue)) < EPSILON)
+					sprintf(buffer, "|%4d", (int) round(shiftValue));
+				else sprintf(buffer, "|%2.2f", shiftValue);
+				rep << buffer;
+				if(Tools::isSunday(day)) rep << "| ";
+			}
+
+			rep << "|" << std::endl;
+		}
+	}
+	rep << std::endl;
+
+	cout << rep.str();
+
+	return rep.str();
+}
+
+string MasterProblem::workedWeekendsToString(bool printInteger){
+	return "";
+}
```

Comment:
Remember why it is empty? Needed or delete?

Reply:
removed in 00d408bf6bca2c11e9fe599d53209b5f158cf674